src/slam_toolbox_sync.cpp

```diff
@@ -42,6 +36,18 @@ void SynchronousSlamToolbox::run()
 {
   rclcpp::Rate r(100);
   while (rclcpp::ok()) {
+    {
+      // `get_current_state` is not thread safe in humble
+      // see. <https://github.com/ros2/rclcpp/issues/1746>
+      // fixed in rolling. <https://github.com/ros2/rclcpp/pull/1756>
+      boost::mutex::scoped_lock lock(get_state_mutex_);
+      auto state_id = get_current_state().id();
```

Comment:
It may actually be necessary here

Reply:
[89d4a127459a934f444459a64eb6ec60b1102c4b]

Since `scan_filter_` is also generated by `on_activate` and reset by `on_deactivate`, I have updated it to use boost's interruption_point like other threads.

---

https://github.com/SteveMacenski/slam_toolbox/blob/89d4a127459a934f444459a64eb6ec60b1102c4b/src/slam_toolbox_common.cpp#L76-L117

https://github.com/SteveMacenski/slam_toolbox/blob/7b213e81ab1a36c1cc05dfc731ec563463be6f60/src/slam_toolbox_sync.cpp#L93-L103